packages/airnode-node/src/handlers/start-coordinator.ts

```diff
@@ -88,12 +95,44 @@ function aggregateApiCalls(state: CoordinatorState) {
 async function executeApiCalls(state: CoordinatorState) {
   const { aggregatedApiCallsById } = state;
 
-  const aggregatedApiCalls = Object.values(aggregatedApiCallsById);
+  const cachedKeys = caching.getKeys('requestId-');
+
+  const filteredUncachedAggregatedApiCalls = pickBy(
+    aggregatedApiCallsById,
+    (_value, key) => !cachedKeys.includes(`requestId-${key}`)
+  );
+
+  const filteredCachedAggregatedApiCalls: RegularAggregatedApiCallWithResponse[] = Object.entries(
+    aggregatedApiCallsById
+  )
+    .filter(([id, _value]) => cachedKeys.includes(`requestId-${id}`))
+    .map(([id, apiCall]) => {
+      const data = caching.getValueForKey(`requestId-${id}`);
+      return {
+        ...apiCall,
+        data,
+        // only previously successful API calls are cached
+        success: true,
+      };
+    });
+
+  const aggregatedApiCalls = Object.values(filteredUncachedAggregatedApiCalls);
+
   const [logs, processedAggregatedApiCalls] = await calls.callApis(aggregatedApiCalls, getWorkerOptions(state));
   logger.logPending(logs);
 
+  processedAggregatedApiCalls
+    .filter((call) => call.success && call.cacheResponses)
+    .forEach((call) => {
+      caching.addKey(`requestId-${call.id}`, (call as RegularApiCallSuccessResponse).data);
+    });
+
+  caching.syncFsASync();
```

Comment:
I think we should not ignore this promise and instead sync the FS synchronously. Also, do we need to call this at all? Isn't the call in the deployer enough? Is this for airnode-client?

Reply:
> Is this for airnode-client?

Yes I believe so, but this raises the question of if it is here, why then do we need to initialize the cache and (for aws) call sync within the deployer? It seems like `caching.init()` and `caching.syncFsSync()` could be removed from there unless I am missing something.

Reply:
Yeah, seems to me as well, but it doesn't hurt to keep them I guess.